packages/lib-sourcify/test/Compilation/VyperCompilation.spec.ts

```diff
@@ -0,0 +1,184 @@
+import { describe, it } from 'mocha';
+import { expect } from 'chai';
+import path from 'path';
+import fs from 'fs';
+import { VyperCompilation } from '../../src/Compilation/VyperCompilation';
+import { vyperCompiler } from '../utils';
+
+describe.only('VyperCompilation', () => {
+  it('should compile a simple Vyper contract', async () => {
+    const contractPath = path.join(
+      __dirname,
+      '..',
+      'sources',
+      'Vyper',
+      'testcontract',
+    );
+    const contractFileName = 'test.vy';
+    const contractContent = fs.readFileSync(
+      path.join(contractPath, contractFileName),
+      'utf8',
+    );
+
+    const compilation = new VyperCompilation(
+      vyperCompiler,
+      '0.3.10+commit.91361694',
+      {
+        language: 'Vyper',
+        sources: {
+          [contractFileName]: {
+            content: contractContent,
+          },
+        },
+        settings: {
+          evmVersion: 'istanbul',
+          outputSelection: {
+            '*': ['evm.bytecode'],
+          },
+        },
+      },
+      {
+        name: contractFileName.split('.')[0],
+        path: contractFileName,
+      },
+    );
+
+    await compilation.compile();
+    expect(compilation.getCreationBytecode()).to.not.be.undefined;
+    expect(compilation.getRuntimeBytecode()).to.not.be.undefined;
+  });
+
+  it('should handle immutable references correctly', async () => {
+    const contractPath = path.join(
+      __dirname,
+      '..',
+      'sources',
+      'Vyper',
+      'withImmutables',
+    );
+    const contractFileName = 'test.vy';
+    const contractContent = fs.readFileSync(
+      path.join(contractPath, contractFileName),
+      'utf8',
+    );
+
+    const compilation = new VyperCompilation(
+      vyperCompiler,
+      '0.4.0+commit.e9db8d9f',
+      {
+        language: 'Vyper',
+        sources: {
+          [contractFileName]: {
+            content: contractContent,
+          },
+        },
+        settings: {
+          evmVersion: 'london',
+          outputSelection: {
+            '*': ['evm.bytecode'],
+          },
+        },
+      },
+      {
+        name: contractFileName.split('.')[0],
+        path: contractFileName,
+      },
+    );
+
+    await compilation.compile();
+    expect(compilation.getImmutableReferences()).to.not.be.empty;
```

Comment:
Also not check for non-emptiness but exact values

Reply:
I still have to refactor the tests, right now I just wanted to no-brainer test as much lines / branches as possible